image-hd.c

```diff
@@ -605,10 +605,16 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 		part->extended = has_extended &&
 			(partition_table_entries >= hd->extended_partition);
 		if (part->extended) {
-			if (!hd->extended_lba)
-				hd->extended_lba = now;
 			now += hd->align;
 			now = roundup(now, part->align);
+			if (!hd->extended_lba) {
+				if(part->offset) {
+					hd->extended_lba = part->offset;
+				}
+				else {
+					hd->extended_lba = now;
+				}
+			}
```

Comment:
This is not quite correct either: the partition offset is rounded up below. And I don't like changing the meaning of `extended_lba`.

Reply:
The reasoning behind this change was to remove the `hd->align` from `p->offset - hd->align - hd->extended_lba`.
It looked like an antipattern because on line 608, the same value was added (`now += hd->align`)
You're right about the rounding.
I took the liberty of modifying the meaning of  `extended_lba`, as it was not used anywhere else.

Reply:
The code is like that, so it's easier to understand, when you come from the Specification. From the EBR Wiki Page:
"Starting sector (`relative_sectors`) = LBA address of next EBR (`p->offset - hd->align`) minus LBA address of extended partition's first EBR (`hd->extended_lba`)".